Formula/g/gstreamer.rb

```diff
@@ -229,9 +214,18 @@ def install
     plugin_dir = lib/"gstreamer-1.0"
     rpath_args = [loader_path, rpath(source: plugin_dir)].map { |path| "-rpath,#{path}" }
     ENV.append "RUSTFLAGS", "--codegen link-args=-Wl,#{rpath_args.join(",")}"
-    inreplace "subprojects/gst-plugins-rs/cargo_wrapper.py",
-              "env['RUSTFLAGS'] = shlex_join(rust_flags)",
-              "env['RUSTFLAGS'] = ' '.join(rust_flags)"
+
+    # Adjust processing of RUSTFLAGS to avoid using shlex, which may mangle our RPATH-related
+    # flags, due to the presence of `$` in $ORIGIN.
+    wrapper_files = %w[
+      subprojects/gst-plugins-rs/cargo_wrapper.py
+      subprojects/gst-devtools/dots-viewer/cargo_wrapper.py
+    ]
+    inreplace wrapper_files do |s|
+      s.gsub!(/shlex\.split\(env\.get\(("RUSTFLAGS"|'RUSTFLAGS'), (""|'')\)\)/,
+              "' '.split(env.get(\"RUSTFLAGS\", \"\"))")
+      s.gsub! "shlex_join(rust_flags)", "' '.join(rust_flags)"
+    end
```

Comment:
There's a whole bunch of missing RPATH warnings on macOS, so I suspect this change broke RPATH handling on macOS, hence the test failures on arm64 macOS.

If we're doing the `inreplace` to avoid mangling `$ORIGIN`, maybe we don't need it on macOS? We use `@loader_path` there.